### ISSUE TITLE
Improve the handling of synonymous names

### ISSUE THREAD

**Issue by octycs** [TUM-Dev/navigatum#30]
Currently, some similar types of rooms have a lot of different names in TUMOnline, e.g.:

"Rechnerraum-Studenten", "Rechnerarbeitsraum", "Rechnerraum", "Computerarbeitsraum", "Computerraum", "Computerpool", "Computer Studenten", "CIP-Raum", "CIP-Pool", "Computerpool", "CAD-Pool", "EDV-Pool", "Rechner-Raum"

Currently, we have synonyms configured for a lot of them in Meilisearch, but too many synonyms slows search down. To optimize this we could reduce the number of terms already in the data processing, e.g. changing rarely used terms such as "Computerpool" to more common such as "CIP-Pool". Then we only need a synonym to map "Computerpool" => "CIP-Pool", but not the other way round.

Additionally, the server might use these synonyms to optimize the display of results, e.g. show it in the subtext.

**Comment by CommanderStorm** [comment]
> *This was generated by AI during triage.*

## Agent Brief

**Category:** enhancement
**Summary:** Reduce Meilisearch synonym-table size by normalising room-type names at data-processing time and switching the remaining synonyms to one-way mappings — **only if** the current bidirectional synonym table is actually slowing search down. Measure first.

**Current behavior:**
All room-type synonym handling lives in `server/src/setup/search_synonyms.yaml` as **bidirectional** synonym groups. The original reporter's example (Feb 2022) still applies verbatim: 13 different German names for "computer room" (`Rechnerraum`, `Computerraum`, `CIP-Pool`, `EDV-Pool`, `CAD-Pool`, …) are each mapped to all 12 others, producing a dense N×N synonym graph. Since 2022 the synonym file has grown, not shrunk — additional bidirectional groups have been added for libraries, lecture halls, meeting rooms, copy/print rooms, kitchens, cooling, waste, etc. The source room-type strings in `data/external/results/rooms_*.csv` are not normalised; the data pipeline carries the raw variants through unchanged.

**Desired behavior:**
Two paths, gated by measurement:

1. **Step 1 (always do this):** benchmark the current search end-to-end latency on a representative query mix. Then benchmark the same mix with the synonym file replaced by a stripped-down one-way version (e.g. all `Computerraum`-variants → `CIP-Pool` only). If the difference is below a meaningful threshold (say, < 10% on p95 latency, no user-visible difference), **stop and close this issue** with the measurement as the closing comment. The premise the issue rests on (synonyms are slowing search) needs to be true today, not in 2022.

2. **Step 2 (only if the benchmark says it's worth it):** consolidate. For each bidirectional synonym group in `search_synonyms.yaml`, pick a canonical name. Add a normalisation step in `data/processors/` that rewrites the rare variants to the canonical name in the indexed `type`/`name` fields. Replace the bidirectional group in `search_synonyms.yaml` with a one-way mapping (`rare-variant: [canonical]`) so user queries with the rare variant still resolve, but the index itself is keyed on the canonical.

**Key interfaces:**

- `server/src/setup/search_synonyms.yaml` — the synonym configuration loaded into Meilisearch at startup.
- `data/processors/` — where the normalisation step would slot in. Choose canonical names that reflect TUMonline's actual most-common spelling, not the agent's preference.
- `data/external/results/rooms_*.csv` and the source-loading processors — these carry the raw room-type strings; the normalisation step should happen after load, not by editing source data.
- The search-relevance / latency tests in `server/src/search_executor/` — extend with a quantitative latency-measurement test (or a separate `cargo bench`) covering the benchmark step.
- Snapshot tests for queries that currently hit the rare variants — verify they still resolve after consolidation.

**Acceptance criteria:**

- [ ] A measurement is recorded on this issue (or in a benchmark file under version control) for current search latency on a representative query mix.
- [ ] A second measurement is recorded for the same mix with a stripped-down one-way synonym set, demonstrating either (a) measurable improvement → proceed to Step 2, or (b) no measurable difference → close the issue and explain.
- [ ] **If Step 2 runs:** the bidirectional synonym groups in `search_synonyms.yaml` are replaced with one-way mappings to a canonical name per group; a data-processor step normalises the indexed values to that canonical name; existing search snapshots still pass; new snapshots cover queries for the rare variants.
- [ ] **If Step 2 runs:** the choice of canonical name per group is documented (one line per group is fine) so future contributors don't reopen the debate.
- [ ] The original reporter's specific case — searching for any of the 13 computer-room variants — still returns the right rooms after the change.

**Out of scope:**

- Adding new synonym groups — this is consolidation only.
- Touching the English ↔ German cross-language groups (`library` ↔ `Bibliothek` family) — those serve a different purpose (cross-language search) and don't fall under the original report.
- Search-relevance weight tuning (see #296).
- Search UI changes.
- Replacing Meilisearch.
